sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -849,6 +849,12 @@ val GPU_COREDUMP_PIPE_PATTERN = conf("spark.rapids.gpu.coreDump.pipePattern")
       .booleanConf
       .createWithDefault(false)
 
+  val INCOMPATIBLE_SHUFFLE_SKIP_CHECK_ORIGIN =
```

Comment:
We can define an abstrct method and override it in a special shim to enable this skip-check.
Otherwise it requires users need to know and set this conf carefully according to Spark distributions.

Reply:
And this is opposite to the direction of zero config.

Reply:
>We can introduce an abstrct method like incompatShuffleOrigin instead and override it in a special shim to enable this skip-check.

Good idea, then we need to post another PR in a customer repo to implement incompatShuffleOrigin.
I prefer your idea. @winningsix Which solution do you like?